projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php

```diff
@@ -246,18 +248,56 @@ public function add_stats_menu() {
 		add_menu_page( __( 'Stats', 'jetpack' ), $menu_title, 'edit_posts', 'https://wordpress.com/stats/day/' . $this->domain, null, 'dashicons-chart-bar', 3 );
 	}
 
+	/**
+	 * Holds the current product, set by get_current_product().
+	 *
+	 * @var array
+	 */
+	private $cached_product = array();
+	/**
+	 * Gets the current product and stores it in $cached_product so the database is only called once per request.
+	 *
+	 * @return array
+	 */
+	private function get_current_product() {
+		// Declare array variable if it does not exist.
+		$current_blog_id = get_current_blog_id();
+		if ( ! isset( $this->cached_product[ $current_blog_id ] ) ) {
+			$this->cached_product[ $current_blog_id ] = null;
+		}
+
+		if ( null === $this->cached_product[ $current_blog_id ] && class_exists( 'WPCOM_Store_API' ) ) {
+			$this->cached_product[ $current_blog_id ] = \WPCOM_Store_API::get_current_plan( get_current_blog_id() );
+		}
+
+		return $this->cached_product[ $current_blog_id ];
+	}
+
+	/**
+	 * Checks if the current product supports WooCommerce installation.
+	 *
+	 * @return bool
+	 */
+	private function current_product_can_install_woocommerce() {
+		if ( class_exists( 'WPCOM_Store' ) ) {
+			$products = \WPCOM_Store::get_wpcom_business_and_higher_plans();
+			$product  = $this->get_current_product();
+			return in_array( $product['product_id'], $products, true );
```

Comment:
re 8-gh-Automattic/woo-on-plans#issuecomment-989168429

So you're thinking put a filter here?

Reply:
I was, but since our conversation, I'm passing the plan to `add_woocommerce_installation_menu` and subsequently the `jetpack_show_wpcom_woocommerce_installation_menu` filter. Then we do the actual plan check in D69877-code. This allows us to compare plans without a new database query, and still gives us the benefit of managing the menu link from wpcom.